sql/api/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -1383,20 +1383,27 @@ class Column(val node: ColumnNode) extends Logging {
   def over(): Column = over(Window.spec)
 
   /**
-   * Marks this column reference as an outer reference for subqueries.
+   * Mark this column as an outer column if its expression refers to columns from an outer query. This
+   * is used to trigger lazy analysis of Spark Classic DataFrame, so that we can use it to build
+   * subquery expressions. Spark Connect DataFrame is always lazily analyzed and does not need to
+   * use this function.
```

Comment:
```suggestion
   * Mark this column as an outer column if its expression refers to columns from an outer query.
   * This is used to trigger lazy analysis of Spark Classic DataFrame, so that we can use it to
   * build subquery expressions. Spark Connect DataFrame is always lazily analyzed and does not
   * need to use this function.
```